File: .idea/app/src/main/java/com/chevstrap/rbx/AppDirectories/RobloxClientData.java
package com.chevstrap.rbx.AppDirectories;

import com.chevstrap.rbx.App;

public class RobloxClientData extends CommonAppData {
    public RobloxClientData() {}

    @Override
    public String getExecutablePackage() {
        String preferred = App.getChevstrapSettings().getPreferredRobloxApp();
        if ("vng".equals(preferred)) {
            return "com.roblox.client.vnggames";
        } else if ("global".equals(preferred)) {
            return "com.roblox.client";
        } else if ("custom".equals(preferred)) {
                return App.getChevstrapSettings().getCustomRobloxClientPackage();
        } else {
            return "com.roblox.client";
        }
    }
}

File: app/src/main/java/com/chevstrap/rbx/appDirectories/RobloxClientData.kt
package com.chevstrap.rbx.appDirectories

import com.chevstrap.rbx.App

class RobloxClientData : CommonAppData() {
    override val executablePackage: String
        get() {
            val preferred: String =
                App.config.data.preferredRobloxApp
            return when (preferred) {
                "vng" -> {
                    "com.roblox.client.vnggames"
                }
                "galaxy_store" -> {
                    "com.roblox.client.samsunggalaxy"
                }
                "global" -> {
                    "com.roblox.client"
                }
                else -> {
                    "com.roblox.client"
                }
            }
        }
}